verilog/tools/kythe/scope_resolver.cc

```diff
@@ -15,12 +15,20 @@
 #include "verilog/tools/kythe/scope_resolver.h"
 
 #include <string>
+#include <vector>
 
 namespace verilog {
 namespace kythe {
 
+bool ScopeMemberItem::operator==(const ScopeMemberItem& other) const {
+  return this->vname == other.vname;
+}
+bool ScopeMemberItem::operator<(const ScopeMemberItem& other) const {
+  return this->vname < other.vname;
+}
+
 void Scope::AddMemberItem(const ScopeMemberItem& member_item) {
-  members_.push_back(member_item);
+  members_.insert(member_item);
```

Comment:
Note: `insert` returns a `pair<iterator, bool>` where the bool is true if insertion succeeded, false if rejected because the element with key already exists.  Should this be signaled to the caller in a return value?  What happens if this code encounters a duplicate definition/declaration?  It should work gracefully, taking a best guess if needed, but might be worth `LOG`-ing.  (And this design choice should be documented.)

Reply:
The members here can be duplicated only because the multi-iterations but they all map to the same member item.